lib/ansible/module_utils/facts/system/systemd.py

```diff
@@ -39,9 +39,16 @@ def collect(self, module=None, collected_facts=None):
             if rc != 0:
                 return systemd_facts
 
-            systemd_facts["systemd"] = {
-                "features": str(stdout.split("\n")[1]),
-                "version": int(stdout.split(" ")[1]),
-            }
+            systemd_version = stdout.split(" ")[1]
+
+            systemd_facts["systemd"] = {}
+            systemd_facts["systemd"]["features"] = stdout.split("\n")[1]
+            systemd_facts["systemd"]["full_version"] = str(
+                stdout.split(" ")[2].split(")")[0][1:],
+            )
+            try:
+                systemd_facts["systemd"]["version"] = int(systemd_version)
+            except ValueError:
+                systemd_facts["systemd"]["version"] = str(systemd_version)
```

Comment:
`systemd_version` is already a string. Also, it's possible to make this look simpler having that dict assignment in just one place:
```suggestion
            with contextlib.suppress(ValueError):
                systemd_version = int(systemd_version)

            systemd_facts["systemd"]["version"] = systemd_version
```

(`import contextlib` at the top)

Reply:
Thanks @webknjaz!
The earlier `str(...)`  is just my being overzealous.

Reply:
`contextlib.suppress` sure made it cleaner, I'll start using that more :)